### ISSUE TITLE
training time

### ISSUE THREAD

**Issue by lwmlyy** [facebookresearch/wsd-biencoders#2]
Hi, I am runing the biencoders with BERT on 2 Tesla v100 but it takes too long time to finish even one epoch (3 and a half hours). I wonder if this is the normal training time or if I have run into some bugs. Thanks for your help.

![biencoders](https://user-images.githubusercontent.com/38210876/85088777-7a0cbf80-b213-11ea-8b49-d491345e621e.png)

**Comment by lwmlyy** [comment]
@blvns

**Comment by blvns** [comment]
Hi! That sounds about right. It took between 2 to 3 hours per epoch (and almost 3 days total) to train the full model on 2 Quadro GP100 GPUS. Did you change the batch sizes from the defaults? That could affect the runtime.

Also, w.r.t. the dev perfromance in your screenshot -- are you using the default learning rate of 1e-4? The best one we found during tuning was 1e-5, which should give better dev performance across the first few epochs.

**Comment by lwmlyy** [comment]
Thanks for your valuable information. It seems v100 runs much slower than GP100 since I used all the default settings. Anyway, that's a lot of work in your research paper, appreciate it.
